runner_manager/models/backend.py

```diff
@@ -184,6 +187,9 @@ def configure_instance(self, runner: Runner) -> AwsInstance:
                 Tags=tags,
             ),
         ]
+        iam_instance_profile: IamInstanceProfileTypeDef = IamInstanceProfileTypeDef(
+            Name=self.iam_instance_profile_name
+        )
```

Comment:
What happens if `iam_instance_profile_name` is not defined and `iam_instance_profile` becomes: `IamInstanceProfileTypeDef(Name="")`, will the EC2 instance be created accordingly?

Reply:
Passing in an empty string (e.g. not setting the value in the config file) successfully creates an EC2 instance without any IAM role associated with it which is what I would expect to happen

Reply:
As far as I can tell :) I have also had a last minute change of heart and changed it from name to ARN as I think thats the better of the two.